sdk/servicebus/service-bus/src/receivers/receiver.ts

```diff
@@ -400,15 +380,12 @@ export class ReceiverImpl<ContextT> implements Receiver<ContextT>, SubscriptionR
       this._context.batchingReceiver = BatchingReceiver.create(this._context, options);
     }
 
-    const messages = await this._context.batchingReceiver.receive(
+    const receivedMessages = await this._context.batchingReceiver.receive(
       maxMessageCount,
       maxWaitTimeInSeconds
     );
 
-    return {
-      messages,
-      context: this.getContext()
-    };
+    return (receivedMessages as unknown) as ReceivedMessageT[];
```

Comment:
This seems odd...
How did we end up casting something as unknown only to cast it to something else right away?

Reply:
It is a bit odd but I believe the issue (and why that particular cast is needed) makes sense.

Essentially, I've got a Receiver<T> where T extends ReceivedMessage|ReceivedLockedMessage (henceforth known as RM|RLM).

I have a concrete class that implements both of those interfaces (ServiceBusMessageImpl).

Here I'm basically saying "ServiceBusMessageImpl implements the proper ReceivedMessageT interface) but TypeScript is saying "it's also possible for 'T' to include derived types from RM|RLM). 

In this case since I control both the instantiation of this class and the choosing of the underlying message type via `ServiceBusClient` I'm not worried. 

I can guarantee that I will meet the letter and law of the contract but the only way to do this cast where TS believes they could be from two separate hierarchies is to cast to unknown and then do the cast to the final object.

Reply:
cc @chradek, @bterlson